### PR TITLE
[Bug 18333] Xcode8 support

### PR DESCRIPTION
**Pull request by livecodepanos**

### CODE REVIEW

**Comment by livecodepanos** [comment]
WIP until we build and upload the prebuilt libs (libcustomcrypto, libcustomssl) for iOS 10.0

**Comment by livecodepanos** [comment]
BTW this PR sets the minimum OS requirements for Xcode 8 **GM** based on the assumption that those will be the same as the ones that currently hold for Xcode 8-**beta**.

If the minimum OS requirements change when Xcode 8 **GM** is released, this PR has to be updated.

**Comment by peter-b** [comment]
@livecodepanos This looks fine to me.  Feel free to mark as reviewed by me as soon as prebuilts are available!

**Comment by montegoulding** [comment]
It would be good at some stage to patch the fix xcode script to optionally copy SDKs to a separate dir so we only need to keep one Xcode on the machine. Mac SSD space is expensive...

**Comment by peter-b** [comment]
@montegoulding Can you file an issue in BZ please?  I'll aim to get that done this sprint.

**Comment by montegoulding** [comment]
@peter-b http://quality.livecode.com/show_bug.cgi?id=18335

**Comment by peter-b** [comment]
@livecodepanos In `setup_xcode_sdks.sh`, in the `min_sdk` function, you need to change `sort -u` to `sort -un`.

**Comment by livecodepanos** [comment]
I am marking this as reviewed so as to have something to test on Monday

@livecode-vulcan review ok ed3ee9b

**Comment by livecode-vulcan** [comment]
💙 review by @livecodepanos ok ed3ee9b1e0f628e71d2a3c36219ab0d5673c884e

**Comment by livecode-vulcan** [comment]
😞 test failure ed3ee9b1e0f628e71d2a3c36219ab0d5673c884e
- try-community-linux-x86: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/164)
- try-community-mac: [pending](https://vulcan.livecode.com/builders/try-community-mac/builds/168)
- try-community-linux-x86_64: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/164)
- try-community-emscripten: [pending](https://vulcan.livecode.com/builders/try-community-emscripten/builds/164)
- try-community-ios: [failure](https://vulcan.livecode.com/builders/try-community-ios/builds/168)
- try-community-win-x86: [pending](https://vulcan.livecode.com/builders/try-community-win-x86/builds/164)
- try-community-android-armv6: [pending](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/164)

**Comment by livecodepanos** [comment]
@livecode-vulcan review ok a9b9070

**Comment by livecode-vulcan** [comment]
💙 review by @livecodepanos ok a9b9070d8de6bcc36692249220a4cf57a7148a43

**Comment by livecode-vulcan** [comment]
😞 test failure a9b9070d8de6bcc36692249220a4cf57a7148a43
- try-community-linux-x86: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/165)
- try-community-mac: [pending](https://vulcan.livecode.com/builders/try-community-mac/builds/169)
- try-community-linux-x86_64: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/165)
- try-community-emscripten: [pending](https://vulcan.livecode.com/builders/try-community-emscripten/builds/165)
- try-community-ios: [failure](https://vulcan.livecode.com/builders/try-community-ios/builds/169)
- try-community-win-x86: [pending](https://vulcan.livecode.com/builders/try-community-win-x86/builds/165)
- try-community-android-armv6: [pending](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/165)

File: ide-support/revdeploylibraryios.livecodescript [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿script "revDeployLibraryIOS"
+script "revDeployLibraryIOS"
```
Comment (montegoulding) [inline ide-support/revdeploylibraryios.livecodescript:1]: Hmm... script only stacks are meant to have a BOM. The engine always writes one when saving.

File: ide-support/revdeploylibraryios.livecodescript [hunk 2]
```diff
@@ -33,20 +33,26 @@ function deployGetIphoneOSes
...
+      // Xcode 8.0 requires a Mac running Mac OS 10.12 or later
+      put "10.0,8.0" into tList[1]
    end if
    
    return tList
 end deployGetIphoneOSes
 
 function deployGetIosMinimumVersions
-   return "5.1.1,6.0,6.1,7.0,7.1,8.0,8.1,8.2,8.3,8.4,9.0,9.1,9.2,9.3"
+   return "5.1.1,6.0,6.1,7.0,7.1,8.0,8.1,8.2,8.3,8.4,9.0,9.1,9.2,9.3,10.0"
```
Comment (montegoulding) [inline ide-support/revdeploylibraryios.livecodescript]: @livecodepanos `5.1.1` needs to be removed here I think

Comment (montegoulding) [inline ide-support/revdeploylibraryios.livecodescript]: Never mind... I somehow got to an outdated diff

**Comment by peter-b** [comment]
@livecode-vulcan retry a9b9070

**Comment by livecode-vulcan** [comment]
😞 test failure a9b9070d8de6bcc36692249220a4cf57a7148a43
- try-community-linux-x86: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/180)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/184)
- try-community-linux-x86_64: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/180)
- try-community-emscripten: [pending](https://vulcan.livecode.com/builders/try-community-emscripten/builds/180)
- try-community-ios: [failure](https://vulcan.livecode.com/builders/try-community-ios/builds/184)
- try-community-win-x86: [pending](https://vulcan.livecode.com/builders/try-community-win-x86/builds/180)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/180)

**Comment by peter-b** [comment]
@livecode-vulcan review ok afe8370

**Comment by livecode-vulcan** [comment]
💙 review by @peter-b ok afe8370f48e4b35921631582a3b7951aea83837d

**Comment by livecode-vulcan** [comment]
😞 test failure afe8370f48e4b35921631582a3b7951aea83837d
- try-community-linux-x86: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/194)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/198)
- try-community-linux-x86_64: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/194)
- try-community-emscripten: [pending](https://vulcan.livecode.com/builders/try-community-emscripten/builds/194)
- try-community-ios: [failure](https://vulcan.livecode.com/builders/try-community-ios/builds/198)
- try-community-win-x86: [pending](https://vulcan.livecode.com/builders/try-community-win-x86/builds/194)
- try-community-android-armv6: [pending](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/194)

**Comment by peter-b** [comment]
@livecode-vulcan review ok 1439dfc

**Comment by livecode-vulcan** [comment]
💙 review by @peter-b ok 1439dfc820b7ff328ee7cda14fac7b17c2dbf689

**Comment by livecode-vulcan** [comment]
😞 test failure 1439dfc820b7ff328ee7cda14fac7b17c2dbf689
- try-community-linux-x86: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/195)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/199)
- try-community-linux-x86_64: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/195)
- try-community-emscripten: [pending](https://vulcan.livecode.com/builders/try-community-emscripten/builds/195)
- try-community-ios: [failure](https://vulcan.livecode.com/builders/try-community-ios/builds/199)
- try-community-win-x86: [pending](https://vulcan.livecode.com/builders/try-community-win-x86/builds/195)
- try-community-android-armv6: [pending](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/195)

**Comment by montegoulding** [comment]
@livecode-vulcan review ok 43c6fc4

**Comment by livecode-vulcan** [comment]
💙 review by @montegoulding ok 43c6fc44fdd5bbf66eb3eb4e2b5b69fab0830879

**Comment by livecode-vulcan** [comment]
😎 test success 43c6fc44fdd5bbf66eb3eb4e2b5b69fab0830879
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/196)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/200)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/196)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/196)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/200)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/196)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/196)

**Pull request merged**